spec.html

```diff
@@ -23791,7 +23791,8 @@ <h1>Runtime Semantics: Evaluation</h1>
             1. Assert: _received_.[[Type]] is ~return~.
             1. Let _return_ be ? GetMethod(_iterator_, *"return"*).
             1. If _return_ is *undefined*, then
-              1. If _generatorKind_ is ~async~, set _received_.[[Value]] to ? Await(_received_.[[Value]]).
+              1. If _generatorKind_ is ~async~, then
+                1. Return Completion Record { [[Type]]: _received_.[[Type]], [[Value]]: ? Await(_received_.[[Value]]), [[Target]]: _received_.[[Target]] }.
               1. Return ? _received_.
```

Comment:
Alternatively, we could change both cases:

```suggestion
              1. Let _value_ be _received_.[[Value]].
              1. If _generatorKind_ is ~async~, then
                1. Set _value_ to ? Await(_value_).
              1. Return Completion Record { [[Type]]: ~return~, [[Value]]: ? Await(_received_.[[Value]]), [[Target]]: ~empty~ }.
```

Reply:
Thank you for the good suggestion. It seems a better way than my suggestion. However, I think you applied `Await` twice to `_value_`. Do you mean like this?
```suggestion
              1. Let _value_ be _received_.[[Value]].
              1. If _generatorKind_ is ~async~, then
                1. Set _value_ to ? Await(_value_).
              1. Return Completion Record { [[Type]]: ~return~, [[Value]]: _value_, [[Target]]: ~empty~ }.
```

Reply:
I applied this suggestion!